packages/payment-processor/test/payment/erc20-escrow-payment.test.ts

```diff
@@ -0,0 +1,331 @@
+        BigNumber.from(payeeAfterBalance).eq(BigNumber.from(payeeBeforeBalance).add(100)),
+      ).toBeTruthy();
+    });
+    it('Should initiate emergency claim', async () => {
+      // Set a new requestID to test independent unit-tests.
+      const request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      request.requestId = 'aacc';
+
+      // Assign the paymentAddress as the payee.
+      const payee = getSigner(provider, paymentAddress);
+
+      // Execute payEscrow.
+      expect(
+        await (await payEscrow(request, wallet, undefined, undefined)).wait(1),
+      ).toBeTruthy();
+
+      // Payer initiate emergency claim.
+      const tx = await initiateEmergencyClaim(request, payee);
+      const confirmedTx = await tx.wait(1);
+      
+      // Checks the status and tx.hash.
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+    });
+    it('Should revert emergency claim', async () => {
+      // Set a new requestID to test independent unit-tests.
+      const request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      request.requestId = 'aadd';
+
+      // Assign the paymentAddress as the payee.
+      const payee = getSigner(provider, paymentAddress);
+
+      // Execute payEscrow.
+      await (await payEscrow(request, wallet, undefined, undefined)).wait(1);
+
+      // Payer initiate emergency claim.
+      await (await initiateEmergencyClaim(request, payee)).wait(1);
+      
+      // Payer reverts the emergency claim.
+      const tx = await revertEmergencyClaim(request, wallet);
+      const confirmedTx = await tx.wait(1);
+
+      // Checks the status and tx.hash.
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+    });
+    it('Should freeze funds:', async () => {
+      // Set a new requestID to test independent unit-tests.
+      const request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      request.requestId = 'aaee';
+
+      // Execute payEscrow.
+      await (await payEscrow(request, wallet, undefined, undefined)).wait(1);
+
+      // Payer freeze escrow funds.
+      const tx = await freezeRequest(request, wallet);
+      const confirmedTx = await tx.wait(1);
+
+      // Checks the status and tx.hash.
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+    });
+    it('Should revert if tried to withdraw to early:', async () => {
+      // Set a new requestID to test independent unit-tests.
+      const request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      request.requestId = 'aaff';
+
+      // Execute payEscrow.
+      await (await payEscrow(request, wallet, undefined, undefined)).wait(1);
+
+      // Payer executes a freeze of escrow funds.
+      await (await freezeRequest(request, wallet)).wait(1);
+      
+      // Expects the transaction to revert.
+      await expect(refundFrozenFunds(request, wallet)).rejects.toThrowError();
```

Comment:
> // Expects the transaction to revert.

Explain what you test instead, don't rephrase the code

```suggestion
      await expect(refundFrozenFunds(request, wallet)).rejects.toThrowError(
        'here the error you expect',
      );
```

That's to be sure we don't throw for buggy test mocks reasons, for instance.

Reply:
Thank you, I'll fix that.

Reply:
```suggestion
      // Payer tries to withdraw frozen funds before unlock date.
      await expect(refundFrozenFunds(request, wallet)).rejects.toThrowError('Not Yet!',);
```